### DISCUSSION TITLE
Sample code for checking disconnected from Bluetooth device

### DISCUSSION THREAD

**Discussion by snowuyl** [hbldh/bleak#1469] [Q&A]
Does bleak provide API or callback function for detecting disconnected from Bluetooth device?

**Comment by dlech** [comment]
https://github.com/hbldh/bleak/blob/develop/examples/disconnect_callback.py

**Reply by MoritzSpiller** [reply]
Hi, I followed the example to implement a disconnect callback. My current problem is that this callback is only called when I'm disconnecting by calling BleakClient.disconnect(), not if my peripheral device disconnects from the computer running bleak. I'm on MacOS 14.2.1, I will test that on Windows and maybe Linux in January. In the meantime, do you have any idea by what this could be caused?

**Reply by dlech** [reply]
The disconnect callback example is working for me on macOS 14.2.1 so we would need more info to reproduce the problem.
